### ISSUE TITLE
Autosave when downloading profiles

### ISSUE THREAD

**Issue by christoarmani** [instaloader/instaloader#1256]
**Provide us a use case of the feature**
How could the user invoke the new function? Which problem would it solve? If new information is obtained, how would it be further processed?
I would like to see instaloader automatically saves an iterator file much like when I manually try to close/stop the process. This is helpful to reduce the number of requests if an error happens or is suddenly closed. 

**Describe the solution you'd like**
Instaloader automatically saves an iterator file let's say after 50 or 100 posts and updates the progress which can help with large profiles.

**Describe alternatives you've considered**
Currently, this is sort of implemented. I have to ctrl + c in order to stop the download of posts to have instaloader to save an iterator file. 

**If the feature request is accepted, would you be willing to submit a pull request?**
Yes / No

Yes with help

**Additional context**
Add any other context about the feature request here.

**Comment by Deleora** [comment]
Are you asking for something different than https://github.com/instaloader/instaloader/pull/732 ?

**Comment by christoarmani** [comment]
I believe that #732 has already been implemented. This is different than #732 since it wouldn't require a manual stop to the iterating process to create the .json file.

**Comment by Deleora** [comment]
So what I understand is you are requesting this feature so that it helps in downloading large profiles?

**Comment by christoarmani** [comment]
Yes, or while downloading multiple profiles from an args.txt.

**Comment by Deleora** [comment]
Have you tried `--latest-stamps` feature?

**Comment by christoarmani** [comment]
I have read the new --latest-stamps feature but this appears to be more for updating than downloading profiles. There have been numerous times when I am unable to manually stop instaloader (appears to get hung up and refuse to save an iterator file) which results in requiring me to restart the profile download from the beginning. I am trying to reduce duplicate requests for profile post downloads.

**Comment by Deleora** [comment]
Okay so have you tried to download any large profiles yet? And if yes, how do you do it without the feature you are requesting?

**Comment by christoarmani** [comment]
Yes, I currently have to Ctrl + C after every 50-100 posts. Which requires me to actively watch the download process. I don't know the exact number of requests Instagram will accept, I generally know after 1000 I should save and stop for another day. 

But there has been times when Instaloader does get hung up and can't properly close and save an iterator file. I am unable to consistently reproduce it but I do know once I get an error about "not Instagram while downloading" I am unable to Ctrl + C.

**Comment by Thammus** [comment]
Like described in #998 it would be very complex to proper implement the resume logic if the post on which the resumable iterator gets created differs from the one Instaloader gets aborted on, especially for features like `--fast-update`.
If I understand it correctly, having the feature you suggested would still require you to manually "Ctrl + C" after 1000 posts. As an alternative, you could use Instaloader as a module and adjust the [rate controller](https://instaloader.github.io/module/instaloadercontext.html#instaloader.RateController) to only allow 1000 requests per 24 hours, and you could also easily [save the iterator](https://instaloader.github.io/module/nodeiterator.html) as often as you like.

**Comment by christoarmani** [comment]
@Thammus Thanks! That makes more sense to do. I appreciate the response.
